feat: v0 standards article (#28)
ethereum/zkvm-website: src/content/blog/zkevm-standards-v0-release.md

---
title: "Ethereum zkVM Interoperability: zkVM Standards v0"  
date: "2026-02-16"  

> excerpt: "The first release of zkvm-standards establishes three initial standards for zkVM development: RV64IM target architecture, C interfaces for precompiles, and standardized IO handling."  

Reviewer: ```suggestion
excerpt: "The first release of zkVM-standards establishes three initial standards for zkVM development: RV64IM target architecture, C interfaces for precompiles, and standardized IO handling."  
```
to match the capitalization used in the rest of the post

author: "Kevaundray Wedderburn, Marcin Bugaj"  
topics: ["standards", "interoperability", "zkvm"]
---

> Imagine an Ethereum execution client team preparing for zkVM-based proving. They face a fragmented landscape: each zkVM requires custom compilation targets, different precompile interfaces, and unique IO handling. Supporting three zkVMs means potentially three separate integrations.

Reviewer: ```suggestion
Imagine an Ethereum execution client team preparing for zkVM-based proving. They face a fragmented landscape: each zkVM requires custom compilation targets, different precompile interfaces, and unique IO handling. Supporting three zkVMs potentially means three separate integrations.
```

By standardizing on RV64IM+Zicclsm, we establish a common compilation baseline that RISC-V zkVMs can target. Each zkVM still has implementation-specific details like memory layouts, however, a shared ISA foundation means optimizations and tooling improvements can benefit the entire ecosystem rather than being siloed per RISC-V zkVM.

### C Interface for Precompiles

Ethereum's EVM has a set of precompiles like BLAKE2F, MODEXP, BLS12-381, etc. Most if not all are expensive to prove using standard instruction traces, so zkVMs implement them as optimized **zkVM precompiles**.

This standard defines a C-based API that allows guest programs to call these zkVM precompiles. This allows execution client teams to write precompile-calling code once and switch between zkVM backends without invasive refactors.

### C interface for IO

Guest programs need to communicate with the *outside* world: receiving private inputs and committing to public outputs that verifiers can check. The IO interface standardizes how this communication happens.

> This standard defines a C-based API for the IO interface, providing two key functions for accessing private input data provided by the host, and committing to public outputs that verifiers can check. 

Reviewer: ```suggestion
This standard defines a C-based API for the IO interface. It provides two key functions for accessing private input data provided by the host, and committing to public outputs that verifiers can check. 
```

Subtly, the interface assumes the region of memory being used to hold the input will not change during program execution.

## What's Next

> **For zkVM implementers**: Adopt these standards and help us refine them. Open issues in the repository or on the [Ethereum R&D Discord](https://discord.com/invite/qGpsxSA) for anything that doesn't fit your architecture.

Reviewer: Should there be a link here for "in the repository"? Or should the relevant implementers have that repo link already? Defer to you on what's best.

Author: Added a link
